crates/matrix-sdk-crypto/src/session_manager/sessions.rs

```diff
@@ -247,11 +225,21 @@ impl SessionManager {
         let mut missing: BTreeMap<_, BTreeMap<_, _>> = BTreeMap::new();
         let mut timed_out: BTreeMap<_, BTreeSet<_>> = BTreeMap::new();
 
-        // Add the list of devices that the user wishes to establish sessions
-        // right now.
-        for user_id in users.filter(|u| !self.failures.contains(u.server_name())) {
-            let user_devices = Box::pin(self.get_user_devices(user_id)).await?;
+        let unfailed_users = users.filter(|u| !self.failures.contains(u.server_name()));
 
+        // XXX: You'd think this would be unnecessary, but if you omit it, Rust starts
```

Comment:
What might be related is the generic `users` possibly not implementing `Send`. Maybe adding `+ Send` to the `impl Iterator` would fix the issue.

Reply:
(as per discussion on Matrix, this does not help)